### DISCUSSION TITLE
downtime when change deployment name

### DISCUSSION THREAD

**Discussion by g-traveller** [argoproj/argo-cd#17641] [Q&A] [answered]
Hi, 

I change the deployment name in kustomize and submit to Gitlab, ArgoCD will delete existing deployment and create new one (I guess deployment name is the metadata in K8S), it will results in a downtime durning the new pod starting.

Is there way to avoid downtime?

Thanks

**Comment by jgwest** [comment, marked as answer]
There are solutions that you could use at the Kubernetes level to avoid downtime, for example, using a load balancer to redirect traffic to the new Deployment before deleting the old Deployment. (See Argo Rollouts for an automated example of this)

But in general this is not a problem that Argo CD attempts to solve.
